rocky/assets/js/renameReports.js

```diff
@@ -1,35 +1,46 @@
 import { onDomReady } from "../js/imports/utils.js";
 
-onDomReady(initConfirmButtons);
-
-export function initConfirmButtons(element) {
-  let root = element || document;
-  let confirm_buttons = root.querySelectorAll(".confirm-button");
-
-  confirm_buttons.forEach((button) => initClickHandlers(button));
-}
-
-function initClickHandlers(button) {
-  button.addEventListener("click", function (event) {
-    const target = event.target.closest("dialog");
+onDomReady(initRenameReports);
+
+function initRenameReports() {
+  initResetButtons();
+
+  const table = document.getElementById("report-name-table");
+  if (!table) return;
+
+  table.addEventListener("click", function (event) {
+    if (!event.target.matches(".reset-button")) return;
```

Comment:
The reset buttons are now handled by this delegated event handler, right? So `initResetButtons` is no longer needed?

Reply:
Since storing the initial `value` for later use doesn't necessarily involve the reset buttons, the name might be confusing. And I wonder if it would make sense to use the [`dataset`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset) to store the initial value instead of a non-standard property on the element instance. Maybe something like...

```js
Array.prototype.forEach.call(table.querySelectorAll('.name-input'), function (input) {
  input.dataset.defaultValue = input.value;
});
```

...and...

```diff
-input.value = input.defaultvalue;
+input.value = input.dataset.defaultvalue;
```

?

Reply:
Thanks for your effort to make this as slick as it is now :)